File: samples/uwp/SetOfCppModules/Printer.cpp
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

#include "pch.h"
#include "Printer.h"

using namespace SetOfCppModules;
using namespace Platform;
using namespace concurrency;

Printer::Printer()
{
}

void Printer::Create(Microsoft::Azure::IoT::Gateway::MessageBus ^bus, Windows::Foundation::Collections::IMapView<Platform::String^, Platform::String^>^ configuration)
{
	std::wstring formattedText = L"SetOfCppModules::Printer::Create\r\n";
	OutputDebugString(formattedText.c_str());
}

void Printer::Destroy()
{
	std::wstring formattedText = L"SetOfCppModules::Printer::Destroy\r\n";
	OutputDebugString(formattedText.c_str());
}

void Printer::Receive(Microsoft::Azure::IoT::Gateway::Message ^received_message)
{
	auto contentBytes = received_message->GetContent();
	std::wstring content;
	for (unsigned int i = 0; i < contentBytes->Size; i++)
	{
		content += (wchar_t)contentBytes->GetAt(i);
	}

	auto props = received_message->GetProperties();

	std::wstring propsString;
	for (auto iter = begin(props); iter != end(props); iter++)
	{
		propsString += (*iter)->Key->Data();
		propsString += L"=";
		propsString += (*iter)->Value->ToString()->Data();
		propsString += L", ";
	}

	std::wstring output = L"SetOfCppModules.Printer.Receive: Content=";
	output += content.c_str();
	output += L"\r\n";
	output += L"SetOfCppModules.Printer.Receive: Properties={";
	output += propsString.c_str();
	output += L"}\r\n";
	OutputDebugString(output.c_str());
}

File: samples/uwp/SetOfCsModules/Printer.cs
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Microsoft.Azure.IoT.Gateway;

namespace SetOfCsModules
{
    public sealed class Printer : IGatewayModule
    {
        public void Create(MessageBus bus, IReadOnlyDictionary<string, string> configuration)
        {
            System.Diagnostics.Debug.WriteLine("SetOfCsModules.Printer.Create");
        }

        public void Destroy()
        {
            System.Diagnostics.Debug.WriteLine("SetOfCsModules.Printer.Destroy");
        }

        public void Receive(Message received_message)
        {
            var contentBytes = received_message.GetContent();
            StringBuilder content = new StringBuilder();
            for (int i = 0; i < contentBytes.Count; i++)
            {
                char ch = (char)contentBytes[i];
                if (ch != '\0')
                {
                    content.Append(ch);
                }
            }

            var props = received_message.GetProperties();
            StringBuilder propsString = new StringBuilder();
            foreach (var key in props.Keys)
            {
                propsString.Append(key);
                propsString.Append("=");
                propsString.Append(props[key] as string);
                propsString.Append(", ");
            }

            StringBuilder sb = new StringBuilder();
            sb.Append("SetOfCsModules.Printer.Receive: Content=" + content.ToString() + "\r\n");
            sb.Append("SetOfCsModules.Printer.Receive: Properties={" + propsString.ToString() + "}\r\n");
            System.Diagnostics.Debug.Write(sb.ToString());
        }
    }
}
